sections/performance/block-loop.md

```diff
@@ -0,0 +1,73 @@
+
+## Example
+Let's take a look at an example from [Node Clinic](https://clinicjs.org/documentation/doctor/05-fixing-event-loop-problem).
+```
+const restify = require('restify')
+const server = restify.createServer()
+
+function sleep (ms) {
+  const future = Date.now() + ms
+  while (Date.now() < future);
+}
+
+server.get('/', function (req, res, next) {
+  sleep(30)
+  res.send({})
+  next()
+})
+
+server.listen(3000)
+```
+
+And when we benchmark this app, we start to see the latency caused by the long
+while loop.
+
+### Run the benchmark 
+`clinic doctor --on-port 'autocannon localhost:$PORT' -- node slow-event-loop`
+
+### The results
+
+```
+─────────┬────────┬────────┬────────┬────────┬───────────┬──────────┬───────────┐
+│ Stat    │ 2.5%   │ 50%    │ 97.5%  │ 99%    │ Avg       │ Stdev    │ Max       │
+├─────────┼────────┼────────┼────────┼────────┼───────────┼──────────┼───────────┤
+│ Latency │ 270 ms │ 300 ms │ 328 ms │ 331 ms │ 300.56 ms │ 38.55 ms │ 577.05 ms │
+└─────────┴────────┴────────┴────────┴────────┴───────────┴──────────┴───────────┘
+┌───────────┬─────────┬─────────┬─────────┬────────┬─────────┬───────┬─────────┐
+│ Stat      │ 1%      │ 2.5%    │ 50%     │ 97.5%  │ Avg     │ Stdev │ Min     │
+├───────────┼─────────┼─────────┼─────────┼────────┼─────────┼───────┼─────────┤
+│ Req/Sec   │ 31      │ 31      │ 33      │ 34     │ 32.71   │ 1.01  │ 31      │
+├───────────┼─────────┼─────────┼─────────┼────────┼─────────┼───────┼─────────┤
+│ Bytes/Sec │ 4.65 kB │ 4.65 kB │ 4.95 kB │ 5.1 kB │ 4.91 kB │ 151 B │ 4.65 kB │
+```
+
+## Image of the Event Loop
+   ┌───────────────────────────┐
+┌─>│           timers          │
+│  └─────────────┬─────────────┘
+│  ┌─────────────┴─────────────┐
+│  │     pending callbacks     │
+│  └─────────────┬─────────────┘
+│  ┌─────────────┴─────────────┐
+│  │       idle, prepare       │
+│  └─────────────┬─────────────┘      ┌───────────────┐
+│  ┌─────────────┴─────────────┐      │   incoming:   │
+│  │           poll            │<─────┤  connections, │
+│  └─────────────┬─────────────┘      │   data, etc.  │
+│  ┌─────────────┴─────────────┐      └───────────────┘
+│  │           check           │
+│  └─────────────┬─────────────┘
+│  ┌─────────────┴─────────────┐
+└──┤      close callbacks      │
+   └───────────────────────────┘
+
+>Here's a good rule of thumb for keeping your Node server speedy: Node is fast when the work associated with each client at any given time is "small".
+>[Don't Block the Event Loop (or the Worker Pool) \| Node.js](https://nodejs.org/en/docs/guides/dont-block-the-event-loop/)
+
```

Comment:
Try viewing the file - the formatting is broken, also see other pages for a reference

Reply:
This should be fixed now. Viewing in VScode or online works.